pkg/handlers/webhookreceiver.go

```diff
@@ -148,14 +148,27 @@ func (h *WebhookReceiverHandler) processAlert(alert template.Alert, mnl *oav1alp
 	}
 	// Send the servicelog for the alert
 	log.WithFields(log.Fields{LogFieldNotificationName: notification.Name}).Info("will send servicelog for notification")
-	err = ocm.BuildAndSendServiceLog(
-		ocm.NewServiceLogBuilder(notification.Summary, notification.ActiveDesc, notification.ResolvedDesc, viper.GetString(config.ExternalClusterID), notification.Severity, notification.LogType, notification.References),
-		firing, &alert, h.ocm)
-	if err != nil {
-		log.WithError(err).WithFields(log.Fields{LogFieldNotificationName: notification.Name, LogFieldIsFiring: true}).Error("unable to send a notification")
-		metrics.SetResponseMetricFailure("service_logs")
-		metrics.CountFailedServiceLogs(notification.Name)
-		return err
+
+	var retrycount int = 0
+	for {
+		slerr := ocm.BuildAndSendServiceLog(ocm.NewServiceLogBuilder(notification.Summary, notification.ActiveDesc, notification.ResolvedDesc, viper.GetString(config.ExternalClusterID), notification.Severity, notification.LogType, notification.References), firing, &alert, h.ocm)
+		if slerr == nil {
+			break
+		} else {
+			log.WithError(err).WithFields(log.Fields{LogFieldNotificationName: notification.Name, LogFieldIsFiring: true}).Error("unable to send a notification")
+			retrycount++
+			time.Sleep(10*time.Second + 5*time.Second*time.Duration(time.Duration(math.Pow(2, float64(retrycount))).Seconds()))
+			if retrycount == 7 {
```

Comment:
Overall, I think having time based retry limit would be better than count based. Initially a static timeout would be fine too (Eg. 5 minutes). We can still have exponential retries in that interval. Eventually the timeout could be configurable as well and can be easier for us to manage in tests than always having to wait for 7 retries even in tests.